### ISSUE TITLE
Problems with Haval addition to dynamic

### ISSUE THREAD

**Issue by magnumripper** [magnumripper/JohnTheRipper#1514]
OSX
```
dynamic_preloads.c:164:33: warning: '_Funcs_87' defined but not used [-Wunused-variable]
  static DYNAMIC_primitive_funcp _Funcs_##N##7[] = { DynamicFunc__clean_input2_kwik, DynamicFunc__append_salt2, DynamicFunc__append_input2_from_input, DynamicFunc__##H##_crypt_input2_to_output1_FINAL, NULL }; \
                                 ^
dynamic_preloads.c:1339:1: note: in expansion of macro 'DYNA_PRE_DEFINE_LARGE_HASH'
 DYNA_PRE_DEFINE_LARGE_HASH(SHA512,8,128)
 ^
dynamic_preloads.c:165:33: warning: '_Funcs_88' defined but not used [-Wunused-variable]
  static DYNAMIC_primitive_funcp _Funcs_##N##8[] = { DynamicFunc__clean_input2_kwik, DynamicFunc__##H##_crypt_input1_append_input2, DynamicFunc__append_input2_from_input2, DynamicFunc__##H##_crypt_input2_to_output1_FINAL, NULL };
                                 ^
dynamic_preloads.c:1339:1: note: in expansion of macro 'DYNA_PRE_DEFINE_LARGE_HASH'
 DYNA_PRE_DEFINE_LARGE_HASH(SHA512,8,128)
 ^
```

**Comment by magnumripper** [comment]
I think the warnings look a lot alarming than is actually the case. It's just a couple of functions not yet used, right? Then I guess this will be fixed shortly and we can close this.

**Comment by jfoug** [comment]
Yes, a couple unused.

I worked hard trying to get everything working.  This set of macros was NOT easy to get there.  Surprisingly, that CPP in visual C is much more robust than in gcc.  But I got things figured out.

In the section that lays out the array of items for the actual dynamic formats, if built for SIMD-64, the dyna-77/78 get defined out and they were not done in a macro.  But the macro that drops the functions out I did use a macro for sha512.  I will simply have to remove the sha512 from using the macro, or make a shorter macro skipping the last 2 function for a SIMD64 build.
